### ISSUE TITLE
On current dulwich HEAD, Pushing a tag to GitHub deletes all other tags

### ISSUE THREAD

**Issue by Julian** [jelmer/dulwich#441]
Steps to reproduce, given a fresh (empty) GitHub repository called `$repo`:

```
repo='git@github.com:Julian/test.git'
repo_name=test
git clone "$repo"
cd "$repo_name"
git commit --allow-empty -m "Test"
git push origin master
git tag -a created-locally -m "A message"
git push --tags
python -c "from dulwich import porcelain, repo; repo = repo.Repo('.'); porcelain.tag_create(repo, 'dulwich', annotated=True, tag_time=1467737730); porcelain.push(repo, '$repo', refspecs=['refs/tags/dulwich'])"
```

on the latest release, both tags are present (on say the releases page), but on latest HEAD, an error is printed to stderr:

```
/Users/Julian/.local/share/virtualenvs/floodgate/lib/python2.7/site-packages/dulwich/porcelain.py:601: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
  b" failed -> " + e.message.encode(err_encoding) +
Push to git@github.com:Julian/test.git failed -> refs/heads/master failed to update
```

and all the tags other than the one dulwich created are deleted.

(Obviously apologies if this turns out to be an issue with GH's server implementation)

**Comment by jelmer** [comment]
Thanks a lot for the detailed bug report. This ought to have been caught by the testsuite. I'm pushing a fix + test to master now.

**Comment by jelmer** [comment]
Fixed in 0.14.1.
